sunpy/coordinates/wcs_utils.py

```diff
@@ -36,7 +41,7 @@ def solar_wcs_frame_mapping(wcs):
     ycoord = wcs.wcs.ctype[1][0:4]
 
     if xcoord == 'HPLN' and ycoord == 'HPLT':
```

Comment:
That's more of a loaded question than it seems.

The short version is that they don't need them to represent the frame properly. Only HPC needs rsun and only HPC and HCC need observer location.

How `astropy.coordinates` works is that `SkyCoord` keeps track of these attributes (so if you go HPC->HG->HPC they will be set correctly, the idea of the frames is that they are self-contained and you only worry about round tripping and persistence of the attributes at the `SkyCoord` layer.